sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -472,6 +511,54 @@ void Scheduler::cleanupCommands(const std::vector<Command *> &Cmds) {
   }
 }
 
+void Scheduler::deferMemObjRelease(const std::shared_ptr<SYCLMemObjI> &MemObj) {
+  {
+    std::lock_guard<std::mutex> Lock{MDeferredMemReleaseMutex};
+    MDeferredMemObjRelease.push_back(MemObj);
+  }
+  cleanupDeferredMemObjects(/*Blocking*/ false);
+}
+
+inline bool Scheduler::isDeferredMemObjectsEmpty() {
+  std::lock_guard<std::mutex> Lock{MDeferredMemReleaseMutex};
+  return MDeferredMemObjRelease.empty();
+}
+
+void Scheduler::cleanupDeferredMemObjects(bool Blocking) {
```

Comment:
Very minor. Sorry for suggesting it so late, but we can reuse 
`enum BlockingT` here.

Reply:
fixed in https://github.com/intel/llvm/pull/6837/commits/fdab0e78a2f82d394d531c1e5ff80fc5e2197848